### ISSUE TITLE
Fix S2259 FP: Reset fields on this invocation with flow captures

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6103]
#6065 introduced "reset" logic for fields, if an instance method is invoked. The logic fails, if the invocation takes an argument expression, that branches:
```cs
if (this.ObjectField == null)
{{
    this.InstanceMethod(StaticObjectField == null ? 1 : 0);
}}
this.ObjectField.ToString();  // Don't report here
```

Reproducer: #6102
